File: Development3.0/CoreAutomation/Core_Automation/Core_Automation_Mar_14/GW/regression/Process_creation/Finetuned/CSVConnector.cs
///////////////////////////////////////////////////////////////////////////////////////////////////
//
// This file is part of the  R A N O R E X  Project.
// http://www.ranorex.com
//
///////////////////////////////////////////////////////////////////////////////////////////////////

using System;
using System.Collections.Generic;
using System.Text;
using System.Data;

/// <summary>
/// Represents a data connector to a CSV-file.
/// </summary>
class CSVConnector
{
    private string fileName = null;
    private DataTable dt = null;

    /// <summary>
    /// Initializes a new instance of the <see cref="CSVConnector"/> class.
    /// </summary>
    /// <param name="fileName">The path to the CSV file.</param>
    public CSVConnector(String fileName)
    {
        if (fileName == null)
            throw new ArgumentNullException("fileName");

        this.fileName = fileName;
        dt = new DataTable();
        ParseCSVData();
    }

    /// <summary>
    /// Gets the data rows.
    /// </summary>
    public DataRowCollection Rows
    {
        get { return dt.Rows; }
    }

    /// <summary>
    /// Gets the data columns.
    /// </summary>
    public DataColumnCollection Header
    {
        get { return dt.Columns; }
    }

    private void ParseCSVData()
    {
        try
        {
            String[] csvData = System.IO.File.ReadAllLines(fileName);

            if (csvData.Length == 0)
                return;

            String[] headings = csvData[0].Split(';');

            foreach (string header in headings)
            {
                dt.Columns.Add(header, typeof(string));
            }
            
            for (int j = 1; j < csvData.Length; j++)
            {
                DataRow row = dt.NewRow();

                for (int i = 0; i < headings.Length; i++)
                {
                    row[i] = csvData[j].Split(';')[i];
                }
                dt.Rows.Add(row);
            }
        }
        catch (Exception ex)
        {
            throw new DataException("Failed to parse CSV file '" + fileName + "'.", ex);
        }
    }
}

File: Development3.0/CoreAutomation/Core_Automation/Core_Automation_Mar_14/GW/Scripts_regression/UI_TC/UI_TC/CSVConnector.vb
'''////////////////////////////////////////////////////////////////////////////////////////////////
'
' This file is part of the  R A N O R E X  Project.
' http://www.ranorex.com
'
'''////////////////////////////////////////////////////////////////////////////////////////////////

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Data

''' <summary>
''' Represents a data connector to a CSV-file.
''' </summary>
Class CSVConnector
	Private fileName As String = Nothing
	Private dt As DataTable = Nothing

	''' <summary>
	''' Initializes a new instance of the <see cref="CSVConnector"/> class.
	''' </summary>
	''' <param name="fileName">The path to the CSV file.</param>
	Public Sub New(fileName As [String])
		If fileName Is Nothing Then
			Throw New ArgumentNullException("fileName")
		End If

		Me.fileName = fileName
		dt = New DataTable()
		ParseCSVData()
	End Sub

	''' <summary>
	''' Gets the data rows.
	''' </summary>
	Public ReadOnly Property Rows() As DataRowCollection
		Get
			Return dt.Rows
		End Get
	End Property

	''' <summary>
	''' Gets the data columns.
	''' </summary>
	Public ReadOnly Property Header() As DataColumnCollection
		Get
			Return dt.Columns
		End Get
	End Property

	Private Sub ParseCSVData()
		Try
			Dim csvData As [String]() = System.IO.File.ReadAllLines(fileName)

			If csvData.Length = 0 Then
				Return
			End If

			Dim headings As [String]() = csvData(0).Split(";"C)

			For Each header As String In headings
				dt.Columns.Add(header, GetType(String))
			Next

			For j As Integer = 1 To csvData.Length - 1
				Dim row As DataRow = dt.NewRow()

				For i As Integer = 0 To headings.Length - 1
					row(i) = csvData(j).Split(";"C)(i)
				Next
				dt.Rows.Add(row)
			Next
		Catch ex As Exception
			Throw New DataException("Failed to parse CSV file '" & fileName & "'.", ex)
		End Try
	End Sub
End Class
